doc/vim.md

```diff
@@ -0,0 +1,341 @@
+# Using VIM under Talon
+
+This document serves as an instruction manual and quick tutorial for people
+wanting to set up running vim under talon.
+
+Most of the testing has been done on Linux, but if you do test this on other
+systems and you have positive or negative results you can provide feedback on
+the slack channel.
+
+Please note that if you want to have the full vim experience you will have to
+make modifications to both the vim config and talon.
+
+Currently supported VIM features:
+
+* motions
+* registers
+* macros
+* folds
+* tabs
+* splits
+* plugins (see list below)
+* settings
+* automatic mode switching (including terminal)
+
+You can contact fidget on the Talon slack for questions/support.
+
+# Initial Setup Walkthrough
+
+## Talon Change - The word `yank`
+
+The default Talon alphabet uses `yank` for the letter `y`. This conflicts with
+the natural VIM verb, so it is recommended you change the `y` word in
+`code/keys.py`. For example you could instead use: `yell`
+
+## Talon Change - The key `end`
+
+The `code/vim.py` script include support for a motion verb called `end`. To the
+problem is that talon also supports the ability to simply say the word `end` and
+it will trigger pressing the keyboard key `end`.
+
+You currently have two options to deal with this:
+
+1) You can disable the `end` key in talon, however this will negatively impact
+other scenarios where you would normally like to be able to press key directly.
+
+2) Use the alternate vim motion verb `end word`. There aren't too many
+downside to this approach aside from it being somewhat cumbersome.
+
+## Talon Change - The key `word`
+
+By default talon will use the command `word` as a command for saying a single
+word. See `misc/formatters.talon`. In vim "word" is a natural movement motion
+who it is included by default. If you decide to use this you might want to
+change the talon command to be a separate word.
+
+## The `generic_editor.talon` commands
+
+The default actions defined in `generic_editor.talon` are too simple for more
```

Comment:
Would it make sense to disable generic_editor.talon instead?

`not tag: vim`

Reply:
ah ya, I will try that out. originally I just wanted to get rid of errors in a rush so i just nuked it, but ya.. perhaps there's some other way to also integrate with it even better but I just haven't had time to look yet